src/engine/UIManager.cxx

```diff
@@ -287,7 +304,10 @@ void UIManager::toggleGroupVisibility(const std::string &groupID, UIElement *sen
   if (sender)
   {
     Button *button = dynamic_cast<Button *>(sender);
-
+    if (groupID == "SettingsMenu" && !(sender->getUiElementData().text == "Cancel" || sender->getUiElementData().text == "OK"))
```

Comment:
we should add const string / macro for those strings (SettingsMenu/Cancel...), it could be modified in the future, it would be hard to search and change all of them.

Reply:
the entire UIManager will be gone with the new UI update, so i didn't put too much effort in those changes tbh. just ignore the code there for now.